website/project/views/contributor.py

```diff
@@ -505,6 +507,32 @@ def send_claim_email(email, user, node, notify=True, throttle=24 * 3600):
     return to_addr
 
 
+@contributor_added.connect
+def notify_contributor(node, contributor, throttle=24 * 3600):
+    if contributor.is_registered:
+        contributor_record = node.contributor_record.get(contributor._id, {})
+        if contributor_record:
+            timestamp = contributor_record.get('last_sent', None)
+            if timestamp:
+                if not throttle_period_expired(timestamp, throttle):
+                    raise HTTPError(400, data=dict(
+                        message_long='User can only be added to a project once every {} hours'.format(throttle / 3600)
+                    ))
+        else:
+            node.contributor_record[contributor._id] = {}
+
+        mails.send_mail(
+            contributor.username,
+            mails.CONTRIBUTOR_ADDED,
+            user=contributor,
+            node=node,
+            sharing_page=node.web_url_for('node_contributors', _absolute=True)
```

Comment:
Ok, there are cases where we use `add_contributor` but we do not want the signal sent, like when a registration is being created. But I can probably disconnect the signal in those cases.